### PR TITLE
feat: add clear-button-visible API from <vaadin-text-field>

### LINKED ISSUE #105: Add clearButtonVisible API
After combo-box 5.0.0, time-picker needs to be updated to depend on that version AND add the API for having the clear button visible

### PR DESCRIPTION
**Pull request by web-padawan**
Fixes #105

---
This change is [Reviewable](https://reviewable.io/reviews/vaadin/vaadin-time-picker/107)

### CODE REVIEW

File: test/input-test.html [hunk 1]
```diff
@@ -222,6 +222,29 @@
...
+          textField = timePicker.__inputElement;
+          clearButton = textField.$.clearButton;
+        });
+
+        it('should propagate clear-button-visible attribute to text-field', () => {
+          timePicker.clearButtonVisible = true;
+          expect(textField).to.have.property('clearButtonVisible', true);
+        });
+
+        it('clear button should be hidden when disabled', () => {
```
[the remark this answers is not in the document]
Comment (Haprog) [reply test/input-test.html]: I'm not sure why I didn't realize this when I wrote these tests for `vaadin-combo-box`. 😅
These test descriptions are not blockers.

[the remark this answers is not in the document]
Comment (web-padawan) [reply test/input-test.html]: Renamed the tests.

File: test/input-test.html [hunk 2]
```diff
@@ -222,6 +222,29 @@
...
+          timePicker.disabled = true;
+          expect(getComputedStyle(clearButton).display).to.equal('none');
+        });
+
+        it('clear button should be hidden when read-only', () => {
```
[the remark this answers is not in the document]
Comment (web-padawan) [reply test/input-test.html]: Renamed.

**Comment by Haprog** [comment]
Regarding my notes above about buggy behaviour:

1. is probably caused by a bug in `vaadin-combo-box-light` (I will check and create an issue).
2. seems to be caused by a bug in `vaadin-text-field` (I just reported this here https://github.com/vaadin/vaadin-text-field/issues/365)

**Pull request merged**